### PR TITLE
docker_container: don't parse/interpret options if state is 'absent'

### LINKED ISSUE #45486: docker_container state=absent fails when networks are absent too
##### SUMMARY
If you are deleting a container which has some networks set up in place, if the networks are absent already, it fails.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

docker_container

##### ANSIBLE VERSION

```paste below
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/yajo/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, May 16 2018, 17:50:09) [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

Travis, targeting docker hosts.

##### STEPS TO REPRODUCE

```yaml
docker_network:
  name: mynet
  state: absent

docker_container:
  name: mycontainer
  networks:
  - name: mynet
  state: absent
```

##### EXPECTED RESULTS

When setting the container state to `absent`, ansible shouldn't care if networks do not exist.

##### ACTUAL RESULTS
Even if `mycontainer` is absent and this one is the 2nd playbook loop, the `docker_container` task will fail.

```paste below
failed: [localhost] (item={'ansible_connection': 'docker', 'inventory_file': '/role/tests/inventory.cfg', 'inventory_dir': '/role/tests', 'test_playground_tag': 'ubuntu-16.04', 'inventory_hostname': 'ubuntu-16.04', 'inventory_hostname_short': 'ubuntu-16', 'group_names': ['test-playground'], 'ansible_facts': {}, 'test_playground_round': '2', 'playbook_dir': '/role/tests', 'ansible_playbook_python': '/usr/bin/python3.6', 'groups': {'all': ['localhost', 'centos-7', 'debian-8', 'debian-9', 'ubuntu-16.04', 'ubuntu-18.04'], 'ungrouped': [], 'test-playground-controller': ['localhost'], 'test-playground': ['centos-7', 'debian-8', 'debian-9', 'ubuntu-16.04', 'ubuntu-18.04']}, 'omit': '__omit_place_holder__1ffc96311c3e8cb35b9caec8fb05ca18b107cdcf', 'ansible_version': {'string': '2.6.4', 'full': '2.6.4', 'major': 2, 'minor': 6, 'revision': 4}, 'ansible_check_mode': False, 'ansible_diff_mode': False, 'ansible_forks': 5, 'ansible_inventory_sources': ['/role/tests/inventory.cfg'], 'ansible_skip_tags': [], 'ansible_run_tags': ['all'], 'ansible_verbosity': 1}) => {"changed": false, "item": {"ansible_check_mode": false, "ansible_connection": "docker", "ansible_diff_mode": false, "ansible_facts": {}, "ansible_forks": 5, "ansible_inventory_sources": ["/role/tests/inventory.cfg"], "ansible_playbook_python": "/usr/bin/python3.6", "ansible_run_tags": ["all"], "ansible_skip_tags": [], "ansible_verbosity": 1, "ansible_version": {"full": "2.6.4", "major": 2, "minor": 6, "revision": 4, "string": "2.6.4"}, "group_names": ["test-playground"], "groups": {"all": ["localhost", "centos-7", "debian-8", "debian-9", "ubuntu-16.04", "ubuntu-18.04"], "test-playground": ["centos-7", "debian-8", "debian-9", "ubuntu-16.04", "ubuntu-18.04"], "test-playground-controller": ["localhost"], "ungrouped": []}, "inventory_dir": "/role/tests", "inventory_file": "/role/tests/inventory.cfg", "inventory_hostname": "ubuntu-16.04", "inventory_hostname_short": "ubuntu-16", "omit": "__omit_place_holder__1ffc96311c3e8cb35b9caec8fb05ca18b107cdcf", "playbook_dir": "/role/tests", "test_playground_round": "2", "test_playground_tag": "ubuntu-16.04"}, "msg": "Parameter error: network named ansible_test_playground could not be found. Does it exist?"}
```

You can [see the error in this public travis build](https://travis-ci.org/Tecnativa/ansible-test-playground/builds/427158190#L939), from where you have links to go to the source code and see the whole role, playbook, etc.

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Don't parse and/or interpret options (ports, networks, IP addresses, ...) when `state` is `'absent'`. Fixes #45486.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
docker_container

##### ANSIBLE VERSION
```
2.7.0
```

### CODE REVIEW

**Comment by felixfontein** [comment]
Tested successfully with https://github.com/felixfontein/ansible-docker_container-test

**Comment by felixfontein** [comment]
I added a regression test for this: https://github.com/felixfontein/ansible-docker_container-test/compare/regression-45700 I'll merge it once this PR is merged.

**Review by kassiansun — APPROVED** [review]

**Comment by kassiansun** [comment]
shipit

**Comment by felixfontein** [comment]
Thanks for reviewing, @kassiansun!

**Pull request merged**
